app/models/mixins/lifecycle_mixin.rb

```diff
@@ -0,0 +1,21 @@
+module LifecycleMixin
+  extend ActiveSupport::Concern
+
+  def update_lifecycle_state
+    case miq_request.request_state
+    when "finished"
+      lifecycle_state = miq_request.status == 'Ok' ? "provisioned" : "error_provisioned"
```

Comment:
@lfu should error_provisioned be error_in_provisioning? or can it be just error

Reply:
@mkanoor We had a meeting with @Fryguy where we discussed possibly using the lifecycle_state for retirement in the future, so we're trying to use 'provision' in the state where possible.

Reply:
I will take ```error_in_provisioning``` to support retirement in the future.